src/traces/ohlc/helpers.js

```diff
@@ -0,0 +1,110 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+var Lib = require('../../lib');
+
+// This routine gets called during the trace supply-defaults step.
```

Comment:
@bpostlethwaite basically all the hackery I needed is in this file.

I hope you find the comments useful.